packages/core/js-client/src/compilerSupport/registerService.ts

```diff
@@ -14,66 +14,65 @@
  * limitations under the License.
  */
 
-import type { ServiceDef, ServiceImpl } from "@fluencelabs/interfaces";
-
 import { FluencePeer } from "../jsPeer/FluencePeer.js";
 import { logger } from "../util/logger.js";
 
 import { registerGlobalService, userHandlerService } from "./services.js";
+import { ServiceImpl } from "./types.js";
 
 const log = logger("aqua");
 
 interface RegisterServiceArgs {
   peer: FluencePeer;
-  def: ServiceDef;
   serviceId: string | undefined;
   service: ServiceImpl;
 }
 
+const findAllPossibleRegisteredServiceFunctions = (
+  service: ServiceImpl,
+): Set<string> => {
+  let prototype: Record<string, unknown> = service;
+  const serviceMethods = new Set<string>();
+
+  do {
+    Object.getOwnPropertyNames(prototype)
+      .filter((prop) => {
+        return typeof prototype[prop] === "function" && prop !== "constructor";
+      })
+      .forEach((prop) => {
+        return serviceMethods.add(prop);
+      });
+
+    // Satisfying typescript here
```

Comment:
Would be better if these comments have a bit more substance. Even without this comment it's totally clear that you are satifing typescript. It doesn't add anything. Better to explain why are you sure the assertion is correct and will never fail (e.g. you own the objects they are not comming from user, you checked previously somewhere etc.)

Reply:
here I believe you don't actually need to satisfy typescript if you remove type definition from `let prototype: Record<string, unknown> = service` and leave it just with `let prototype = service`

Reply:
There is no way to do this hack without assertion

Reply:
Added more informative description